### PR TITLE
DENA-671: validate tiered storage

### PR DESCRIPTION
**Pull request by sbuliarca**
Added validations for tiered storage:
- for a retention period longer than 3 days enable tiered storage & the local.retention.ms is specified

I'd like to do a new release after this, as this is the most common & complex validation that we're doing

### CODE REVIEW

File: rules/msk_topic_config.go [hunk 1]
```diff
@@ -305,48 +305,179 @@ func (r *MSKTopicConfigRule) validateCleanupPolicy(
...
-	if !hasRt {
-		msg := fmt.Sprintf("%s must be defined on a topic with cleanup policy delete", retentionTimeAttr)
+	retentionTime, err := r.getAndValidateRetentionTime(runner, config, configKeyToPairMap)
+	if err != nil || retentionTime == nil {
+		return err
+	}
+
+	if *retentionTime <= tieredStorageThresholdInDays*millisInOneDay && !isInfiniteRetention(*retentionTime) {
+		return nil
+	}
```
Comment (matthewhughes-uw) [inline rules/msk_topic_config.go:341]: nit/style: could be more explicit in separating error handling from the logic flow (the check could also possibly be bundled with the following `if` to put all the logic in one place: if `retentionTime == nil || ( ... )`)

```suggestion
	if err != nil {
		return err
	}

	if retentionTime == nil {
		return nil
	}
	if *retentionTime <= tieredStorageThresholdInDays*millisInOneDay && !isInfiniteRetention(*retentionTime) {
		return nil
	}
```

**Review by matthewhughes-uw — APPROVED** [review]

File: rules/msk_topic_config.go (hunk 1 above)
Comment (sbuliarca) [reply rules/msk_topic_config.go:341]: Yea.. I had my doubts about this one too when writing it.. I'll split it

Comment (sbuliarca) [reply rules/msk_topic_config.go]: Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/22/commits/a6c1fd9dccb0507d7e4f60ca4ea3e04a97cf899e

File: rules/msk_topic_config.go [hunk 2]
```diff
@@ -305,48 +305,179 @@ func (r *MSKTopicConfigRule) validateCleanupPolicy(
...
+}
+
+func (r *MSKTopicConfigRule) validateTieredStorageEnabled(
+	runner tflint.Runner,
+	config *hclext.Attribute,
+	configKeyToPairMap map[string]hcl.KeyValuePair,
+) error {
+	tieredStoragePair, hasTieredStorageAttr := configKeyToPairMap[tieredStorageEnableAttr]
+	tieredStorageEnableMsg := fmt.Sprintf(
+		"tiered storage should be enabled when retention time is longer than %d days",
```
[the remark this answers is not in the document]
Comment (sbuliarca) [reply rules/msk_topic_config.go]: Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/22/commits/a6c1fd9dccb0507d7e4f60ca4ea3e04a97cf899e

**Pull request merged**